### PR TITLE
Feature/119946 rsvp block ux improvement

### PR DESCRIPTION
**Pull request by Camwyn**
🎟  https://central.tri.be/issues/119946 && https://central.tri.be/issues/120365

Edit templates to streamline the process when login is required for RSVPs.
Pass status through login into template and preload form if present.
Fix login url so it redirects correctly.

### CODE REVIEW

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:x: | Expected 1 space before array closer, found 2. | [content.php:21](https://github.com/moderntribe/event-tickets/blob/040572abd5982ce30c89a951995bf34826dc36b8/src/views/blocks/rsvp/content.php#L21)
:x: | Expected 1 space before array closer, found 2. | [content.php:22](https://github.com/moderntribe/event-tickets/blob/040572abd5982ce30c89a951995bf34826dc36b8/src/views/blocks/rsvp/content.php#L22)
:x: | Expected 1 space before array closer, found 2. | [content.php:25](https://github.com/moderntribe/event-tickets/blob/040572abd5982ce30c89a951995bf34826dc36b8/src/views/blocks/rsvp/content.php#L25)
:x: | Don't leave error_log lying around | [form.php:23](https://github.com/moderntribe/event-tickets/blob/040572abd5982ce30c89a951995bf34826dc36b8/src/views/blocks/rsvp/form.php#L23)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not ')' | [form.php:31](https://github.com/moderntribe/event-tickets/blob/040572abd5982ce30c89a951995bf34826dc36b8/src/views/blocks/rsvp/form.php#L31)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$args' | [form.php:31](https://github.com/moderntribe/event-tickets/blob/040572abd5982ce30c89a951995bf34826dc36b8/src/views/blocks/rsvp/form.php#L31)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not 'false' | [form.php:31](https://github.com/moderntribe/event-tickets/blob/040572abd5982ce30c89a951995bf34826dc36b8/src/views/blocks/rsvp/form.php#L31)

> via jenkins/codesniffer-phpcs

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not ')' | [form.php:31](https://github.com/moderntribe/event-tickets/blob/bd3d818bbde9b2b4029b308d7bf78a4f684728d6/src/views/blocks/rsvp/form.php#L31)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$args' | [form.php:31](https://github.com/moderntribe/event-tickets/blob/bd3d818bbde9b2b4029b308d7bf78a4f684728d6/src/views/blocks/rsvp/form.php#L31)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not 'false' | [form.php:31](https://github.com/moderntribe/event-tickets/blob/bd3d818bbde9b2b4029b308d7bf78a4f684728d6/src/views/blocks/rsvp/form.php#L31)

> via jenkins/codesniffer-phpcs

File: src/views/blocks/rsvp/content.php [hunk 1]
```diff
@@ -13,15 +13,15 @@
  * @version 4.9
  *
  */
-
+$going = ! empty( $_GET[ 'going' ] ) ? $_GET[ 'going' ] : '';
```
Comment (sc0ttkclark) [inline src/views/blocks/rsvp/content.php]: Should this also use sanitize_text_field() even if $going is esc_html() in the template itself?

Comment (sc0ttkclark) [reply src/views/blocks/rsvp/content.php]: Is the intention of `going` to be a 1 in these cases? Should this be set accordingly?

Comment (sc0ttkclark) [reply src/views/blocks/rsvp/content.php]: My bad it's `yes` or empty, so yeah let's just go ahead and add sanitize_text_field() here anyway

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not ')' | [form.php:31](https://github.com/moderntribe/event-tickets/blob/3c59758ef05a52a3e6a3cb2da4c8bb14da1a6ff7/src/views/blocks/rsvp/form.php#L31)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$args' | [form.php:31](https://github.com/moderntribe/event-tickets/blob/3c59758ef05a52a3e6a3cb2da4c8bb14da1a6ff7/src/views/blocks/rsvp/form.php#L31)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not 'false' | [form.php:31](https://github.com/moderntribe/event-tickets/blob/3c59758ef05a52a3e6a3cb2da4c8bb14da1a6ff7/src/views/blocks/rsvp/form.php#L31)

> via jenkins/codesniffer-phpcs

File: src/views/blocks/rsvp/content.php (hunk 1 above)
Comment (Camwyn) [reply src/views/blocks/rsvp/content.php]: OK, sanitized - worth noting that that value is via a checkbox, so not _truly_ user input. Still, worth sanitizing anything received through `$_GET`

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not ')' | [form.php:31](https://github.com/moderntribe/event-tickets/blob/aacf3fe86ff2d9acac5416a4dbe8c57ac6575cb2/src/views/blocks/rsvp/form.php#L31)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$args' | [form.php:31](https://github.com/moderntribe/event-tickets/blob/aacf3fe86ff2d9acac5416a4dbe8c57ac6575cb2/src/views/blocks/rsvp/form.php#L31)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not 'false' | [form.php:31](https://github.com/moderntribe/event-tickets/blob/aacf3fe86ff2d9acac5416a4dbe8c57ac6575cb2/src/views/blocks/rsvp/form.php#L31)

> via jenkins/codesniffer-phpcs

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not ')' | [form.php:31](https://github.com/moderntribe/event-tickets/blob/6d67c94e0c93efd6bcf960f4b0d4397aa728fe2b/src/views/blocks/rsvp/form.php#L31)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$args' | [form.php:31](https://github.com/moderntribe/event-tickets/blob/6d67c94e0c93efd6bcf960f4b0d4397aa728fe2b/src/views/blocks/rsvp/form.php#L31)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not 'false' | [form.php:31](https://github.com/moderntribe/event-tickets/blob/6d67c94e0c93efd6bcf960f4b0d4397aa728fe2b/src/views/blocks/rsvp/form.php#L31)

> via jenkins/codesniffer-phpcs

**Comment by Camwyn** [comment]
On hold until after I get strategy sign-off (an hour?)

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not ')' | [form.php:31](https://github.com/moderntribe/event-tickets/blob/8f33539ff24ce0c98de69a3318367797f7d2e2e0/src/views/blocks/rsvp/form.php#L31)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$args' | [form.php:31](https://github.com/moderntribe/event-tickets/blob/8f33539ff24ce0c98de69a3318367797f7d2e2e0/src/views/blocks/rsvp/form.php#L31)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not 'false' | [form.php:31](https://github.com/moderntribe/event-tickets/blob/8f33539ff24ce0c98de69a3318367797f7d2e2e0/src/views/blocks/rsvp/form.php#L31)

> via jenkins/codesniffer-phpcs

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not ')' | [form.php:31](https://github.com/moderntribe/event-tickets/blob/8bb1cbeb0e195ecbbcc763dc914249feda457c49/src/views/blocks/rsvp/form.php#L31)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$args' | [form.php:31](https://github.com/moderntribe/event-tickets/blob/8bb1cbeb0e195ecbbcc763dc914249feda457c49/src/views/blocks/rsvp/form.php#L31)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not 'false' | [form.php:31](https://github.com/moderntribe/event-tickets/blob/8bb1cbeb0e195ecbbcc763dc914249feda457c49/src/views/blocks/rsvp/form.php#L31)

> via jenkins/codesniffer-phpcs

**Pull request merged**